modules/client/src/test/kotlin/jp/co/soramitsu/iroha2/BlockStreamTest.kt

```diff
@@ -38,89 +40,109 @@ class BlockStreamTest : IrohaTest<Iroha2Client>() {
+        var newAssetDefinition = isi[0].cast<InstructionBox.Register>().extractAssetDefinition()
         assertNotNull(newAssetDefinition)
         assertEquals(newAssetName, newAssetDefinition.id.name.string)
         assertEquals(DEFAULT_DOMAIN, newAssetDefinition.id.domainId.asString())
 
-        // get the last block second time
-        blocksResult = client.subscribeToBlockStream(2, 1)
-        blocks = mutableListOf<VersionedBlockMessage>()
-        blocksResult.collect { block -> blocks.add(block) }
-        instructions = checkBlockStructure(
-            blocks[0],
-            2,
-            DEFAULT_DOMAIN,
-            BOB_ACCOUNT,
-            1,
-        )
-        newAssetDefinition = instructions[0].cast<InstructionBox.Register>().extractAssetDefinition()
-        assertNotNull(newAssetDefinition)
-        assertEquals(newAssetName, newAssetDefinition.id.name.string)
-        assertEquals(DEFAULT_DOMAIN, newAssetDefinition.id.domainId.asString())
-    }
+        subscription.unsubscribe()
 
-    private fun getCommittedBlock(versionedBlockMessage: VersionedBlockMessage): CommittedBlock {
-        return versionedBlockMessage.cast<VersionedBlockMessage.V1>()
-            .blockMessage.versionedCommittedBlock.cast<VersionedCommittedBlock.V1>().committedBlock
+//        blocks = mutableListOf()
+//        subscription.receiveBlocking<VersionedBlockMessage>(actionId).collect { block -> blocks.add(block) }
+//        isi = checkBlockStructure(blocks[0], 2, DEFAULT_DOMAIN, BOB_ACCOUNT, 1)
+//
+//        newAssetDefinition = isi[0].cast<InstructionBox.Register>().extractAssetDefinition()
+//        assertNotNull(newAssetDefinition)
+//        assertEquals(newAssetName, newAssetDefinition.id.name.string)
+//        assertEquals(DEFAULT_DOMAIN, newAssetDefinition.id.domainId.asString())
     }
 
-    private fun getInstructionPayload(committedBlock: CommittedBlock): TransactionPayload {
-        return committedBlock.transactions[0]
-            .cast<VersionedValidTransaction.V1>()
-            .validTransaction
-            .payload
+    @Test
+    @WithIroha([DefaultGenesis::class])
+    @Story("Successful subscription to endless block stream")
+    @SdkTestId("subscription_to_endless_block_stream")
+    @ResourceLock("blockStream")
+    fun `subscription to endless block stream`(): Unit = runBlocking {
+        val repeatTimes = 5
+        val shift = 1 // to test not to take more than was ordered
+        val idToSubscription = client.subscribeToBlockStream(
+            onBlock = { block -> block.extractBlock().height() },
+            closeIf = { block -> block.extractBlock().height() == BigInteger.valueOf(repeatTimes.toLong()) },
+        )
+        val initialActionId = idToSubscription.first
+        val subscription = idToSubscription.second
+        var lastHeight = BigInteger.ZERO
+
+        subscription.receive<BigInteger>(initialActionId) { lastHeight = it }
```

Comment:
better to check sum of heights or smth, since it can be treated as a check of last received block only